spec/index.html

```diff
@@ -10187,8 +10191,39 @@ <h3>SPARQL Algebra</h3>
         <p>The <a href="#defn_algReduced" class="algFct">Reduced</a> solution sequence modifier does not guarantee a defined multiplicity.</p>
         <div class="defn">
           <p><b>Definition: <span id="defn_algSlice">Slice</span></b></p>
-          <p>Let <var>Ψ</var> be a sequence of solution mappings. We define:</p>
-          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, <var>start</var>, <var>length</var>)[<var>i</var>] = <var>Ψ</var>[<var>start</var>+<var>i</var>] for <var>i</var> = 0 to (<var>length</var>-1)</p>
+          <p>Let <var>Ψ</var> be a sequence of solution mappings, and |offset| and |length| are integers. We define:</p>
+          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, |offset|)
+            = <var>Ψ</var>
+            &nbsp; if |offset| &leq; 0</p>
+          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, |offset|)
+            = <var>Ψ<sub>0</sub></var>
+            &nbsp; if |offset| &gt; <a href="#defn_Card">Card</a>(<var>Ψ</var>)</p>
+          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, |offset|)
+            = subseq(<var>Ψ</var>, |offset|+1, <a href="#defn_Card">Card</a>(<var>Ψ</var>)&minus;|offset|)
+            &nbsp; if 0 &lt; |offset| &lt; <a href="#defn_Card">Card</a>(<var>Ψ</var>)</p>
+          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, |offset|, |length|)
+            = <var>Ψ</var>
+            &nbsp; if |offset| &leq; 0
+            and |length| &geq; <a href="#defn_Card">Card</a>(<var>Ψ</var>)</p>
+          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, |offset|, |length|)
+            = <var>Ψ<sub>0</sub></var>
+            &nbsp; if |offset| &gt; <a href="#defn_Card">Card</a>(<var>Ψ</var>)
+            or |length| &leq; 0</p>
+          <p><a href="#defn_algSlice" class="algFct">Slice</a>(<var>Ψ</var>, |offset|, |length|)
+            = subseq(<var>Ψ</var>, |offset|+1, <a href="#defn_Card">Card</a>(<var>Ψ</var>)&minus;|offset|)
```

Comment:
I may be missing something, but it's not clear to me why `offset` is incremented with `1` here (and the line below).
I suspect `subseq` considers sequences with zero-indexed elements, so this increment seems unnecessary to me.

Reply:
> No, subseq is one-based

Ok, then this makes sense indeed.

Should we clarify this somewhere that `subseq` is 1-based? (or perhaps say somewhere higher up that sequences are considered 1-based, e.g. near the beginning of 18.6)
But if I'm the only one that tripped over this, then we can probably keep this as-is.